validations/src/validations.rs

```diff
@@ -198,6 +198,7 @@ struct Counter {
 impl Counter {
     /// Increment by one the counter for a particular category.
     fn increment(&mut self, category_id: usize) {
+        // Ensure safe indexing over `self.categories`
         if category_id < self.categories.len() {
```

Comment:
We want to panic if `category_id` is out of bounds, this means the library is being misused.